### DISCUSSION TITLE
How to derive a class from st7789.ST7789 ?

### DISCUSSION THREAD

**Discussion by mhgue** [russhughes/st7789_mpy#148] [General]
The straight forward way I thought is
```
from st7789 import ST7789
class ST7789_ext(ST7789):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
```
But this fails with "TypeError: function doesn't take keyword arguments" but constructing like this works with keywords:
```
 self.display = st7789.ST7789( self.spi, 240, 240, dc=Pin(8, Pin.OUT), reset=Pin(12, Pin.OUT),
                                      cs=Pin(9, Pin.OUT), backlight=Pin(13, Pin.OUT) )
```
Why are keywords not taken if used as a superclass ?

I am using a RP2040 with MicroPython 1.21 and latest st7789 build in.

**Comment by arthurtong** [comment]
Yes, I have this problem too！

**Comment by russhughes** [comment]
I will look into this.

**Comment by ricksorensen** [comment]
This discussion on the micropython site might be pertinent:
https://github.com/orgs/micropython/discussions/10236

**Reply by russhughes** [reply]
Yes, that looks pertinent. Thanks for the pointer.
